packages/core/src/tools/utils/objectUtils.ts

```diff
@@ -1,7 +1,5 @@
-import { assign } from './polyfills'
-
 export function shallowClone<T>(object: T): T & Record<string, never> {
-  return assign({}, object)
+  return { ...object } as T & Record<string, never>
```

Comment:
Somehow, if `T` is `undefined`, then TS will infer `{...object}` as `undefined`, which does not match the JS that will return `{}`

see [TS playground](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBAMwK4DsDGMCWEVwM4AWAhgDYkQDuAwuSsADwAqAfABQQBGAVsBgFxxGASjgBvAFBw4UYDCRRcouADpVnHhjgBfOOK3jxaHHnhE4AXnzEylGjmCtUAE2AJMdJ0PEB6b1LgAegD8BobG8BwWVqTk1LQOoggQEAIA5BxEUKlaXr7+wQZAA)